src/sources/mainwindow.cpp

```diff
@@ -3,35 +3,78 @@
 #include <QFile>
 #include <QStatusBar>
 #include <QFileDialog>
+#include <QDesktopWidget>
+#include <opencv2/opencv.hpp>
 
 MainWindow::MainWindow(QWidget *parent) :
     QMainWindow(parent),
     ui(new Ui::MainWindow)
 {
     ui->setupUi(this);
     this->setWindowTitle(WINDOW_TITLE);
+    //this->showMaximized();
 }
 
 MainWindow::~MainWindow()
 {
     delete ui;
 }
 
-void MainWindow::on_actionLoad_Reference_Image_triggered()
-{
+/**
+ * @brief Method to convert a cv::Mat object to QImage object.
+ * @param cv::Mat object.
+ * @return Non null QImage object.
+ */
+QImage matToQImage(const cv::Mat& mat) {
+    if(mat.type() == CV_8UC1) {
+        // 8-bits unsigned, NO. OF CHANNELS=1
+        //Example would be grayscale images
+
+        // Set the color table (used to translate colour indexes to qRgb values)
+        QVector<QRgb> colorTable;
+        for (int i=0; i<256; i++)
+            colorTable.push_back(qRgb(i,i,i));
+        // Copy input Mat
+        const uchar *qImageBuffer = (const uchar*) mat.data;
+        // Create QImage with same dimensions as input Mat
+        QImage qimg(qImageBuffer, mat.cols, mat.rows, mat.step, QImage::Format_Indexed8);
+        qimg.setColorTable(colorTable);
+        return qimg;
+    } else {
+        return QImage();
+    }
+}
+
+void MainWindow::on_actionLoad_Reference_Image_triggered() {
     QString fileName = QFileDialog::getOpenFileName(this,
                                                      tr("Select Reference Image File"),
                                                      "",
                                                      "Images (*.png *.jpg *.jpeg *.tif *.tiff);;All Files (*)"
                                                      );
     QString status = tr("");
     if (fileName.length() == 0) status = tr("No File Selected");
-    else status = tr("Selected ") + fileName;
-    statusBar()->showMessage(status, 2000);
+    else {
+        status = tr("Selected ") + fileName;
+        statusBar()->showMessage(status, 2000);
+        cv::Mat refImage = cv::imread(fileName.toStdString(), cv::IMREAD_GRAYSCALE);
+        QImage convertedImage = matToQImage(refImage);
+        //reducing image width and height
+        //this code is buggy if image has very high aspect ratio or very low i think
+        //example would be sceenshot of a dual monitor setup
+        //we may not come across such examples tho. so it should be fine
+        int width = convertedImage.width() % 512;
+        int height = convertedImage.height() % 512;
+
+        int minSize = std::min(width, height);
+        minSize = 512 / minSize;
+        height *= minSize;
+        width *= minSize;
```

Comment:
This is incorrect.
What if image is 512 x 800 ?

Reply:
We're scaling down so that max dimension is 512 right?

Reply:
why exactly do we need to scale down?

Reply:
@iniyanakila Sir, This is so that we could see full portion of large image on screen.

Reply:
oh, it is for visualizing the image in the window... then its fine. If you can define a region of interest (ROI) by drawing a rectangle in the reference image and if you can convert the ROI into its 8-bit version, then you immediately proceed to finding intensity gradients and B-spline coefficients for the defined ROI.

PS. If there are any difficult parts that are not important to proceed further, we can come back and write code for that later. Prioritize whenever required.

cheers
Iniyan